post-til-private/src/main/java/no/difi/meldingsutveksling/ptp/MeldingsformidlerClient.java

```diff
@@ -91,39 +100,73 @@ private Dokument dokumentFromDocument(Document document) {
                 .build();
     }
 
+    private SikkerDigitalPostKlient createSikkerDigitalPostKlient(AktoerOrganisasjonsnummer aktoerOrganisasjonsnummer, ClientInterceptor clientInterceptor) {
+        KlientKonfigurasjon klientKonfigurasjon = createKlientKonfigurasjonBuilder().soapInterceptors(clientInterceptor).build();
+
+        return createSikkerDigitalPostKlient(klientKonfigurasjon, aktoerOrganisasjonsnummer);
+    }
+
     private SikkerDigitalPostKlient createSikkerDigitalPostKlient(AktoerOrganisasjonsnummer aktoerOrganisasjonsnummer) {
-        KlientKonfigurasjon klientKonfigurasjon = KlientKonfigurasjon.builder(config.getEndpoint()).connectionTimeout(20, TimeUnit.SECONDS).build();
+        KlientKonfigurasjon klientKonfigurasjon = createKlientKonfigurasjonBuilder().build();
 
+        return createSikkerDigitalPostKlient(klientKonfigurasjon, aktoerOrganisasjonsnummer);
+    }
 
+    private SikkerDigitalPostKlient createSikkerDigitalPostKlient(KlientKonfigurasjon klientKonfigurasjon, AktoerOrganisasjonsnummer aktoerOrganisasjonsnummer) {
         Databehandler tekniskAvsender = Databehandler.builder(aktoerOrganisasjonsnummer.forfremTilDatabehandler(), Noekkelpar.fraKeyStoreUtenTrustStore(keyStore, config.getKeystore().getAlias(), config.getKeystore().getPassword())).build();
 
         return new SikkerDigitalPostKlient(tekniskAvsender, klientKonfigurasjon);
     }
 
+    private KlientKonfigurasjon.Builder createKlientKonfigurasjonBuilder() {
+        return KlientKonfigurasjon.builder(config.getEndpoint()).connectionTimeout(20, TimeUnit.SECONDS);
+    }
+
+
     public ExternalReceipt sjekkEtterKvittering(String orgnr) {
-        SikkerDigitalPostKlient klient = createSikkerDigitalPostKlient(AktoerOrganisasjonsnummer.of(orgnr));
+        Kvittering kvittering = new Kvittering();
+        PayloadInterceptor payloadInterceptor = new PayloadInterceptor(kvittering::setRawReceipt);
+        SikkerDigitalPostKlient klient = createSikkerDigitalPostKlient(AktoerOrganisasjonsnummer.of(orgnr), payloadInterceptor);
         final ForretningsKvittering forretningsKvittering = klient.hentKvittering(KvitteringForespoersel.builder(config.getPriority()).mpcId(config.getMpcId()).build());
         if (forretningsKvittering == null) {
             return EMPTY_KVITTERING;
         }
-        return Kvittering.from(forretningsKvittering).withCallback(klient::bekreft);
+        return kvittering.setEksternKvittering(forretningsKvittering).withCallback(klient::bekreft);
     }
 
 
     public static class Kvittering implements ExternalReceipt {
         private ForretningsKvittering eksternKvittering;
         private Consumer<ForretningsKvittering> callback;
+        private String rawReceipt;
         private final ServiceIdentifier serviceIdentifier = ServiceIdentifier.DPI;
 
+        public Kvittering() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1186)